src/main/java/com/redhat/devtools/lsp4ij/LSPIJEditorUtils.java

```diff
@@ -0,0 +1,419 @@
+        private final String blockCommentPrefix;
+        private final String blockCommentSuffix;
+
+        private LSPIJDefaultCommenter(@NotNull PsiFile file) {
+            LSPClientFeatures clientFeatures = getClientFeatures(file);
+            this.lineCommentPrefix = clientFeatures != null ? clientFeatures.getLineCommentPrefix(file) : null;
+            this.blockCommentPrefix = clientFeatures != null ? clientFeatures.getBlockCommentPrefix(file) : null;
+            this.blockCommentSuffix = clientFeatures != null ? clientFeatures.getBlockCommentSuffix(file) : null;
+        }
+
+        @Override
+        public @Nullable String getLineCommentPrefix() {
+            return lineCommentPrefix;
+        }
+
+        @Override
+        public @Nullable String getBlockCommentPrefix() {
+            return blockCommentPrefix;
+        }
+
+        @Override
+        public @Nullable String getBlockCommentSuffix() {
+            return blockCommentSuffix;
+        }
+
+        @Override
+        public @Nullable String getCommentedBlockCommentPrefix() {
+            return null;
+        }
+
+        @Override
+        public @Nullable String getCommentedBlockCommentSuffix() {
+            return null;
+        }
+
+        @Nullable
+        private static LSPClientFeatures getClientFeatures(@NotNull PsiFile file) {
+            CompletableFuture<List<LanguageServerItem>> languageServersFuture = LanguageServiceAccessor.getInstance(file.getProject()).getLanguageServers(
+                    file.getVirtualFile(),
+                    clientFeatures -> StringUtil.isNotEmpty(clientFeatures.getLineCommentPrefix(file)) ||
+                                      StringUtil.isNotEmpty(clientFeatures.getBlockCommentPrefix(file)) ||
+                                      StringUtil.isNotEmpty(clientFeatures.getBlockCommentSuffix(file)),
+                    null
+            );
+            //noinspection TryWithIdenticalCatches
+            try {
+                waitUntilDone(languageServersFuture, file);
+            } catch (ProcessCanceledException e) {
+                return null;
+            } catch (CancellationException e) {
+                return null;
+            } catch (ExecutionException e) {
+                return null;
+            }
+            if (!isDoneNormally(languageServersFuture)) {
+                return null;
+            }
+
+            List<LanguageServerItem> languageServers = languageServersFuture.getNow(Collections.emptyList());
+            Set<LSPClientFeatures> matchingClientFeatures = languageServers
+                    .stream()
+                    .filter(Objects::nonNull)
+                    .map(LanguageServerItem::getClientFeatures)
+                    .collect(Collectors.toSet());
+            // TODO: If there are multiple, can/should be merge them?
+            return matchingClientFeatures.size() == 1 ? ContainerUtil.getFirstItem(matchingClientFeatures) : null;
```

Comment:
Right now we're only returning client features if there's exactly one for the file with comment information.